cohorts/variant_stats.py

```diff
@@ -181,13 +207,19 @@ def variant_stats_from_variant(variant,
     """
     all_stats = []
     for (variant_file, variant_metadata) in metadata.items():
-        if ".maf" in variant_file.lower():
+        if _vcf_is_maf(variant_file=variant_file):
             stats = maf_somatic_variant_stats(variant, variant_metadata)
-        elif "strelka" in variant_file.lower():
+        elif _vcf_is_strelka(variant_file=variant_file):
             stats = strelka_somatic_variant_stats(variant, variant_metadata)
-        elif "mutect" in variant_file.lower():
+        elif _vcf_is_mutect(variant_file=variant_file):
             stats = mutect_somatic_variant_stats(variant, variant_metadata)
         else:
-            raise ValueError("Cannot parse sample fields, variant file {} is from an unsupported caller.".format(variant_file))
+            try:
```

Comment:
I think the underlying function should throw a specific error type we could catch, in order to avoid this. In general it seems having a greater diversity of specific errors would help avoid this.